Can't build with clang on FreeBSD

<!-- For bugs or other problems please provide the following details in addition to
     your issue report, if applicable. See also https://wxpython.org/pages/how-to-submit-issue/

     For issues about building on Linux, please read this page before reporting it here:
     https://wxpython.org/blog/2017-08-17-builds-for-linux-with-pip/
-->

**Operating system**: FreeBSD
**wxPython version & source**:          wxPython 4.0.6 from github. I'm trying to create a FreeBSD package
**Python version & source**:            python 3.6.9  <!-- stock, anaconda, EDM, distro, self-built, etc. -->

**Description of the problem**:

<!-- if possible please include a small runnable application that demonstrates the problem -->

First error
```
[178/868] Compiling sip/cpp/sip_corewxIcon.cpp
../../../../sip/cpp/sip_corewxString.cpp:53:10: error: wxString conversion can only handle WCHAR wxStrings currently
        #error wxString conversion can only handle WCHAR wxStrings currently
         ^
1 error generated.

Waf: Leaving directory `/usr/local/poudriere/ports/default/x11-toolkits/py-wxPython40/work-py36/wxPython-4.0.7/build/waf/3.6/gtk3'
Build failed
 -> task in '_core' failed with exit status 1 (run with -v to display more information)
Command '"/usr/local/bin/python3.6" /usr/local/poudriere/ports/default/x11-toolkits/py-wxPython40/work-py36/wxPython-4.0.7/bin/waf-2.0.8 --wx_config=wxgtk3u-3.0-config --gtk3 --python="/usr/local/bin/python3.6" --out=build/waf/3.6/gtk3 configure build ' failed with exit code 1.
Finished command: build_py (2m45.725s)
Finished command: build (2m45.725s)
Command '"/usr/local/bin/python3.6" -u build.py build' failed with exit code 1.

```
But wxWidgets is compiled with Gtk3 and unicode support.

By removing, the ifdef on this file and ohters. I can build until this issue:

```
[807/868] Compiling sip/cpp/sip_propgridwxPGChoicesData.cpp
../../../../sip/cpp/sip_propgridwxPGPropArgCls.cpp:320:45: error: reinterpret_cast from 'nullptr_t' to 'wxPGProperty *' is not allowed
            *sipCppPtr = new wxPGPropArgCls(reinterpret_cast< wxPGProperty * >(NULL));
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.

Waf: Leaving directory `/usr/local/poudriere/ports/default/x11-toolkits/py-wxPython40/work-py36/wxPython-4.0.7/build/waf/3.6/gtk3'
Build failed
 -> task in '_propgrid' failed with exit status 1 (run with -v to display more information)
Command '"/usr/local/bin/python3.6" /usr/local/poudriere/ports/default/x11-toolkits/py-wxPython40/work-py36/wxPython-4.0.7/bin/waf-2.0.8 --wx_config=wxgtk3u-3.0-config --gtk3 --python="/usr/local/bin/python3.6" --out=build/waf/3.6/gtk3 configure build ' failed with exit code 1.
Finished command: build_py (13m51.176s)
Finished command: build (13m51.177s)
Command '"/usr/local/bin/python3.6" -u build.py build' failed with exit code 1.
*** Error code 1
```

I can create a CI with Cirrus-CI if you want.
Thank you.

> By removing, the ifdef on this file and ohters. I can build until this issue:
> 
> ```
> [807/868] Compiling sip/cpp/sip_propgridwxPGChoicesData.cpp
> ../../../../sip/cpp/sip_propgridwxPGPropArgCls.cpp:320:45: error: reinterpret_cast from 'nullptr_t' to 'wxPGProperty *' is not allowed
>             *sipCppPtr = new wxPGPropArgCls(reinterpret_cast< wxPGProperty * >(NULL));
>                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 1 error generated.
> 
> Waf: Leaving directory `/usr/local/poudriere/ports/default/x11-toolkits/py-wxPython40/work-py36/wxPython-4.0.7/build/waf/3.6/gtk3'
> Build failed
>  -> task in '_propgrid' failed with exit status 1 (run with -v to display more information)
> Command '"/usr/local/bin/python3.6" /usr/local/poudriere/ports/default/x11-toolkits/py-wxPython40/work-py36/wxPython-4.0.7/bin/waf-2.0.8 --wx_config=wxgtk3u-3.0-config --gtk3 --python="/usr/local/bin/python3.6" --out=build/waf/3.6/gtk3 configure build ' failed with exit code 1.
> Finished command: build_py (13m51.176s)
> Finished command: build (13m51.177s)
> Command '"/usr/local/bin/python3.6" -u build.py build' failed with exit code 1.
> *** Error code 1
> ```

I found the solution. LLVM is happy with `static_cast` instead of `reinterpret_cast`

--- a/etg/propgridiface.py
+++ b/etg/propgridiface.py
@@ -99,7 +99,7 @@ def run():
             return sipGetState(sipTransferObj);
         }
         else if (sipPy == Py_None) {
-            *sipCppPtr = new wxPGPropArgCls(reinterpret_cast< wxPGProperty * >(NULL));
+            *sipCppPtr = new wxPGPropArgCls(static_cast< wxPGProperty * >(NULL));
             return sipGetState(sipTransferObj);
         }
         else {

